qubes-rpc/qubes.StartApp

```diff
@@ -1,25 +1,55 @@
-#!/bin/sh
+#!/usr/bin/python3 --
 
-if [ -z "$1" ]; then
-    echo "This service requires an argument" >&2
-    exit 1
-fi
+import sys, os, pwd
+import qubesdb
+from qubesagent.xdg import launch
 
-app_basename="$1.desktop"
+def main(argv):
+    try:
+        (arg0, arg,) = argv
+    except ValueError:
+        print('This service requires an argument', file=sys.stderr)
+        sys.exit(1)
+    try:
+        xdg_data_home = os.environ['XDG_DATA_HOME']
+    except KeyError:
+        xdg_data_home = os.path.expanduser('~/.local/share')
+        if xdg_data_home[0] != '/':
+            raise OSError('cannot expand ~/.local/share')
+    try:
+        xdg_data_dirs = os.environ['XDG_DATA_DIRS']
+    except KeyError:
+        xdg_data_dirs = ['/usr/local/share','/usr/share']
+    else:
+        if '::' in xdg_data_dirs or \
+                xdg_data_dirs.startswith(':') or \
+                xdg_data_dirs.endswith(':'):
+            print('Empty entries in XDG_DATA_DIRS not allowed')
+            sys.exit(1)
+        xdg_data_dirs = xdg_data_dirs.split(':')
+    paths_to_try = [xdg_data_home] + xdg_data_dirs
```

Comment:
When having this in python, we can use already existing implementation: this whole thing can be replaced by:
```python
from xdg.BaseDirectory import xdg_data_dirs
```
(it includes xdg_data_home at the beginning already)